src/datumaro/cli/commands/explore.py

```diff
@@ -125,20 +140,33 @@ def explore_command(args):
         project.working_tree.config.update(build_tree.config)
         project.working_tree.save()
 
-    # Get query datasetitem through query path
-    if osp.exists(args.query):
-        query_datasetitem = None
-        for dataset in source_datasets:
-            try:
-                query_datasetitem = dataset.get_datasetitem_by_path(args.query)
-            except Exception:
-                continue
-            if not query_datasetitem:
-                break
+    if args.query_img_path:
+        querys = (
+            [args.query_img_path]
+            if not isinstance(args.query_img_path, list)
+            else args.query_img_path
+        )
+        query_datasetitems = []
+        for query_ in querys:
+            query_datasetitem = DatasetItem(
+                id=random.randrange(100, 1000), media=Image.from_file(query_)
+            )
+            query_datasetitems.append(query_datasetitem)
+    elif args.query_item_id:
+        querys = (
+            [args.query_item_id] if not isinstance(args.query_item_id, list) else args.query_item_id
+        )
+        query_datasetitems = []
+        for query in querys:
+            for dataset in source_datasets:
+                query_datasetitem = match_query_subset(query, dataset)
+            query_datasetitems.append(query_datasetitem)
+    elif args.query_str:
+        query_datasetitems = args.query_str
     else:
-        query_datasetitem = args.query
+        raise
```

Comment:
I feel that you can remove these two lines since you added the mutually exclusive groups with `required=true`.